src/runtime/hexagon_host.cpp

```diff
@@ -112,7 +112,13 @@ WEAK int init_hexagon_runtime(void *user_context) {
     // code execution based on the runtime; devices with Hexagon hardware will
     // simply provide conduits for execution on that hardware, while test/desktop/etc
     // environments can instead connect a simulator via the API.
-    void *host_lib = halide_load_library("libhalide_hexagon_host.so");
+    // If we are a windows hosted simulater, then we need to load the dll file.
+    // Otherwise if we are a linux hosted simulator we need the .so.
+    void *host_lib = halide_load_library("libhalide_hexagon_host.dll");
```

Comment:
This really should try libhalide_hexagon_host.so first. That's the path that actually occurs in production on device, this is going to generate a guaranteed failed load library call every time on device.